stdlib/public/core/CollectionAlgorithms.swift.gyb

```diff
@@ -141,6 +144,7 @@ extension MutableCollectionType
 ${partitionDocComment}
   ///
 ${orderingRequirementForComparable}
+  @swift3_migration(message="Slice the collection using the range, and call partition()")
   public mutating func partition(range: Range<Index>) -> Index {
```

Comment:
Actually, there is no direct replacement for this API.  The argument-less partition was removed.  You can emulate the Swift 2 partition() in Swift 3 by providing a closure that compares the parameter against an arbitrary element of the collection (e.g., the first element).